### ISSUE TITLE
paltest_getprocesstimes_test2 misbehaves on NetBSD

### ISSUE THREAD

**Issue by krytarowski** [dotnet/coreclr#3283]
paltest_getprocesstimes_test2 is randomly falling down on NetBSD:

```
$ ./bin/obj/NetBSD.x64.Debug/src/pal/tests/palsuite/threading/GetProcessTimes/test2/paltest_getprocesstimes_test2 
kernel1 { .dwLowDateTime=9410, .dwHighDateTime=0 }
kernel2 { .dwLowDateTime=0, .dwHighDateTime=0 }
Unexpected kernel time value reported.
```

```
$ ./bin/obj/NetBSD.x64.Debug/src/pal/tests/palsuite/threading/GetProcessTimes/test2/paltest_getprocesstimes_test2 
kernel1 { .dwLowDateTime=18240, .dwHighDateTime=0 }
kernel2 { .dwLowDateTime=99900, .dwHighDateTime=0 }
user1 { .dwLowDateTime=0, .dwHighDateTime=0 }
user2 { .dwLowDateTime=21879430, .dwHighDateTime=0 }
```

I'm investigating it.

**Comment by krytarowski** [comment]
> The reason is that the scheduler keeps a precise total cpu time of
> a process but the systime/usertime is only sampled by a much slower
> clock. The ratio systime/usertime is then used to return an approximation
> so that systime and usertime sum up to the total time(*).
>
> As a result it is possible that usertime or systime go backwards.
>
> FreeBSD has implemented a workaround by remembering reported
> usertime/systime values for a process and clamping getrusage()
> against these values. As a result the usertime/systime value
> is usually too high but it is monotonic.
>
> Another approach would be to round down the usertime/systime values,
> by not approximating 'measurments' that do not exist. The result
> would be that usertime/systime values that are too low and don't
> sum up to the total CPU time but are again monotonic.
>
> The approximation is done in kern_resource/calc().
> (*) The sampled interrupt time is also included.

by Michael van Elst

-- http://gnats.netbsd.org/cgi-bin/query-pr-single.pl?number=30115

**Comment by krytarowski** [comment]
I'm going to prepare a patch with a walk-around.

**Comment by krytarowski** [comment]
Fixed with #3285
